iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/truncator/StringTruncator.java

```diff
@@ -0,0 +1,24 @@
+package eu.dnetlib.iis.wf.importer.infospace.truncator;
+
+import java.text.BreakIterator;
+
+
+public class StringTruncator {
+
+    /**
+     * Truncates source strings preserving word boundaries, removing words that would be broke.
+     *
+     * @param source String to be truncated.
+     * @param length Length of the source string to keep.
+     * @return Truncated string.
+     */
+    public static String truncateWithoutWordSplit(String source, int length) {
+        if (source.length() <= length) {
+            return source;
+        }
+
+        BreakIterator atWordInstance = BreakIterator.getWordInstance();
+        atWordInstance.setText(source);
+        return source.substring(0, atWordInstance.preceding(length)).trim();
```

Comment:
Discussed offline: string truncation without word split can potentially remove a large part of source string when the string is short and has a longer word at the truncation index. To avoid this kind of situations we agreed on keeping the word at truncation index. With `BreakIterator` this means using `.following` method and this change is applied here. This will also result in changes in test resources with truncated document metadata.